packages/client/test/test-utils/fake/FakeChain.ts

```diff
@@ -12,9 +14,56 @@ export interface StreamRegistryItem {
     permissions: Multimap<EthereumAddress | PublicPermissionTarget, StreamPermission>
 }
 
+export interface Events {
+    streamAddedToStorageNode: (payload: Omit<StorageNodeAssignmentEvent, 'blockNumber'>) => void
+}
+
 export class FakeChain {
-    readonly streams: Map<StreamID, StreamRegistryItem> = new Map()
-    readonly storageAssignments: Multimap<StreamID, EthereumAddress> = new Multimap()
-    readonly storageNodeMetadatas: Map<EthereumAddress, StorageNodeMetadata> = new Map()
-    readonly erc1271AllowedAddresses: Multimap<EthereumAddress, EthereumAddress> = new Multimap()
+
+    private readonly streams: Map<StreamID, StreamRegistryItem> = new Map()
+    private readonly storageAssignments: Multimap<StreamID, EthereumAddress> = new Multimap()
+    private readonly storageNodeMetadatas: Map<EthereumAddress, StorageNodeMetadata> = new Map()
+    private readonly erc1271AllowedAddresses: Multimap<EthereumAddress, EthereumAddress> = new Multimap()
+    private readonly eventEmitter = new EventEmitter<Events>
+
+    getStream(streamId: StreamID): StreamRegistryItem | undefined {
+        return this.streams.get(streamId)
+    }
+
+    setStream(streamId: StreamID, registryItem: StreamRegistryItem): void {
+        this.streams.set(streamId, registryItem)
+    }
+
+    getStorageAssignments(streamId: StreamID): EthereumAddress[] {
+        return this.storageAssignments.get(streamId)
+    }
+
+    addStorageAssignment(streamId: StreamID, nodeAddress: EthereumAddress): void {
+        this.storageAssignments.add(streamId, nodeAddress)
+        this.eventEmitter.emit('streamAddedToStorageNode', { streamId, nodeAddress })
```

Comment:
Do we need to guard against double add?
```suggestion
        const existedBefore = this.storageAssignments.has(streamId, nodeAddress)
        this.storageAssignments.add(streamId, nodeAddress)
        if (!existedBefore) {
            this.eventEmitter.emit('streamAddedToStorageNode', { streamId, nodeAddress })
        }
```

Reply:
There is a guard in the caller (`FakeStreamStorageRegistry#addStreamToStorageNode`), but better move it to here